### DISCUSSION TITLE
cannot connect px4(WSL2) with Airsim, no TCP connection

### DISCUSSION THREAD

**Discussion by AlexHAHA** [microsoft/AirSim#3787] [Support Q&A]
I followed instructions in https://microsoft.github.io/AirSim/px4_sitl_wsl2.html, and installed everything. But after I use `make px4_sitl_default none_iris` in WSL2 and run Airsim in windows, connection is not establised! 

First, I am confused with the output in make process. In Airsim tutorial Video, after make px4_sitl_default none_iris,  normally can get output as follow:
INFO  [simulator] Waiting for simulator to accept connection on TCP port 4560

However I get results:
![image](https://user-images.githubusercontent.com/8017332/121795102-0e034280-cc40-11eb-9702-f973667fa7e9.png)

```
  PWM_MIN: curr: 1000 -> new: 1075
  GPS_UBX_DYNMODEL: curr: 7 -> new: 6
* SYS_AUTOCONFIG: curr: 1 -> new: 0
INFO  [dataman] Unknown restart, data manager file './dataman' size is 11798680 bytes
PX4 SIM HOST: 172.20.160.1
INFO  [simulator] Waiting for simulator to connect on UDP port 14560
```

How could let PX4 simulator to setup TCP connection!

This is my ROMFS/px4fmu_common/init.d-posix/rcS file:
![image](https://user-images.githubusercontent.com/8017332/121795084-e8763900-cc3f-11eb-9a8b-c4ec547145a5.png)

```
# If PX4_SIM_HOST_ADDR environment variable is empty use localhost.
if [ -z "${PX4_SIM_HOST_ADDR}" ]; then
    echo "PX4 SIM HOST: localhost"
    simulator start -c $simulator_tcp_port
else
    echo "PX4 SIM HOST: $PX4_SIM_HOST_ADDR"
    simulator start -c $PX4_SIM_HOST_ADDR $simulator_tcp_port
fi

# only start the simulator if not in replay mode, as both control the lockstep time
if ! replay tryapplyparams
then
	simulator start -c $simulator_tcp_port
fi
```

**Comment by zimmy87** [comment]
In the else block of your startup script, you need to change `simulator start -c` to `simulator start -t`. See [this line](https://github.com/PX4/PX4-Autopilot/blob/master/src/modules/simulator/simulator.cpp#L83) for how the simulator module's arguments specify it's behavior.

**Reply by AlexHAHA** [reply]
Thanks very much, has been solved.  And the experience to me is: if use WSL2, you should complie with version1.12, I found v1.11.3don't work.  because V1.11.3 don't include the commit(https://github.com/PX4/PX4-Autopilot/commit/292a66ce417c9769e1a7845fbc9b8d5e68e1cf0b)
